contracts/CometCore.sol

```diff
@@ -56,6 +56,13 @@ abstract contract CometCore is CometConfiguration, CometStorage, CometMath {
     /// @dev The scale for factors
     uint64 internal constant FACTOR_SCALE = 1e18;
 
+    /// @dev The storage slot for reentrancy guard flags
+    bytes32 internal constant COMET_REENTRANCY_GUARD_FLAG_SLOT = bytes32(keccak256("comet.reentrancy.guard"));
+
+    /// @dev The reentrancy guard status
+    uint256 internal constant COMET_REENTRANCY_GUARD_NOT_ENTERED = 1;
+    uint256 internal constant COMET_REENTRANCY_GUARD_ENTERED = 2;
+
```

Comment:
Can just inline rather than creating constants. Also, let's use a default value of 0. Otherwise, you need to initialize the slot to 1.

Reply:
I recalled seeing in openzeppelin's reentrancy guard contract saying that using 1, 2 is cheaper on running than setting it to 0.  (reference: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/ReentrancyGuard.sol) 

Since we're only checking on if status == COMET_REENTRANCY_GUARD_ENTERED.  It looks like the initialization value of 0 or 1 wouldn't affect the first run behavior, so I ended up changing to 1,2.  And each run it only checks on if the status is ENTERED(2) :P.   But yeah I agree the first initialization value having to be 0 instead of 1 is a bit confusing, and I can change it back to 0 if the gas saving gain is not enough to tradeoff with the confusion initialization values.